### PR TITLE
BRS for  Azure_DBforPostgreSQL_DP_Use_Secure_TLS_Version cotnrol

### PR DESCRIPTION
**Pull request by rajatgoyalMS**
Added BRS for  Azure_DBforPostgreSQL_DP_Use_Secure_TLS_Version cotnrol.

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 1]
```diff
@@ -0,0 +1,847 @@
+<###
+# Overview:
+    This script is used to set minimium required TLS version and enable ssl enforcement for Azure Database for PostgreSQL single server in a Subscription.
+# Control ID:
+    Azure_DBforPostgreSQL_DP_Use_Secure_TLS_Version
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:5]: NIT: Check for such new lines and maintain consistency across the file

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+# Display Name:
+    Use approved version of TLS for Azure Database for PostgreSQL - single Server.
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:7]: NIT: casing single Server

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 3]
```diff
@@ -0,0 +1,847 @@
...
+    
+# Prerequisites:    
+    Contributor or higher priviliged role on the Azure Database for PostgreSQL single server(s) is required for remediation.
+# Steps performed by the script:
+    To remediate:
+        1. Validating and installing the modules required to run the script and validating the user.
+        2. Get the list of Azure Database for PostgreSQL single server(s) in a Subscription that have server parameter minimaltlsversion set as versions less than minimum required TLS version or parameter SslEnforcement is set to OFF .
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:14]: NIT: Minimum TLS Version

Comment (vaishnavipulluri) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:14]: NIT: Spacing at the end

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 4]
```diff
@@ -0,0 +1,847 @@
...
+        2. Get the list of Azure Database for PostgreSQL single server(s) in a Subscription, the changes made to which previously, are to be rolled back.
+        3. Set the server parameter minimaltlsversion to original value and set server parameter SslEnforcement to ON as per input file.
+# Instructions to execute the script:
+    To remediate:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to set supported TLS version for Azure Database for PostgreSQL single server(s) in the Subscription. Refer `Examples`, below.
+    
+    After script execution: 
+        As minimaltlsversion is Static parameter that needs server restart for updated value to take effect, server restart is recommended to be done seperately after script execution.
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:28]: Does it really require restart? Did we verify?

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 5]
```diff
@@ -0,0 +1,847 @@
+<###
+# Overview:
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:2]: Remediation.zip file is not there.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 6]
```diff
@@ -0,0 +1,847 @@
...
+    This script is used to set minimium required TLS version and enable ssl enforcement for Azure Database for PostgreSQL single server in a Subscription.
+# Control ID:
+    Azure_DBforPostgreSQL_DP_Use_Secure_TLS_Version
+# Display Name:
+    Use approved version of TLS for Azure Database for PostgreSQL - single Server.
+    
+# Prerequisites:    
+    Contributor or higher priviliged role on the Azure Database for PostgreSQL single server(s) is required for remediation.
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:10]: Typo - priviliged

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 7]
```diff
@@ -0,0 +1,847 @@
...
+# Steps performed by the script:
+    To remediate:
+        1. Validating and installing the modules required to run the script and validating the user.
+        2. Get the list of Azure Database for PostgreSQL single server(s) in a Subscription that have server parameter minimaltlsversion set as versions less than minimum required TLS version or parameter SslEnforcement is set to OFF .
+        3. Back up details of Azure Database for PostgreSQL single server(s) that are to be remediated.
+        4. Set supported minimum required TLS version by updating parameter minimaltlsversion as minimum required TLS version and enable SslEnforcement by updating server parameter SslEnforcement to ON for Azure Database for PostgreSQL single server(s).
+    To roll back:
+        1. Validate and install the modules required to run the script and validating the user.
+        2. Get the list of Azure Database for PostgreSQL single server(s) in a Subscription, the changes made to which previously, are to be rolled back.
+        3. Set the server parameter minimaltlsversion to original value and set server parameter SslEnforcement to ON as per input file.
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:20]: In rollback why we will set - set server parameter SslEnforcement to ON

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 8]
```diff
@@ -0,0 +1,847 @@
...
+                Write-Host "ERROR: Input file: [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+
+            Write-Host "Fetching all Azure Database for PostgreSQL single server(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $DBForPostgreSQLResources = Import-Csv -LiteralPath $FilePath
+
+            $validDBForPostgreSQLSSResources = $DBForPostgreSQLResources | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:308]: Here we are not checking if validDBForPostgreSQLSSResources is empty then?

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 [hunk 9]
```diff
@@ -0,0 +1,847 @@
...
+
+function  CheckIfOnlySecureTLSVersionConfigured {
+    param (
+        [String] $TLSVersion 
+    )
+    #Check if TLS Versions less than minimum required TLS versions are supported
+    $supportedTLSVersions = @();
+    $nonCompliantTLSVersions = @();
+    try {
+        $supportedTLSVersions = $TLSVersion.ToLower() -replace [Constants]::TLSversionPattern, '' -replace '_', '.';
```
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:797]: we are getting a single TLS value as input, then why are we converting into array, also $supportedTLSVersions is not appropriate name here.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 5 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:2]: added

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 1 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:6]: fixed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 8 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1:311]: checking in the further code as explained.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 2 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1]: done.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 3 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1]: fixed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 4 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1]: it doesn't required restart, removed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 6 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1]: fixed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 7 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1]: fixed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1 (hunk 9 above)
Comment (rajatgoyalMS) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLSingleServerMinReqTLSVersion.ps1]: removed array and made checking only 1 variable.

**Review by vaishnavipulluri — APPROVED** [review]

**Pull request merged**